builder/osc/bsu/step_create_omi.go

```diff
@@ -5,98 +5,91 @@ import (
+	resp, _, err := oscconn.Api.ImageApi.CreateImage(oscconn.Auth).CreateImageRequest(createOpts).Execute()
+	if err != nil || resp.GetImage().ImageId == nil {
 		err := fmt.Errorf("Error creating OMI: %s", err)
 		state.Put("error", err)
 		ui.Error(err.Error())
 		return multistep.ActionHalt
 	}
 
-	image := resp.Image
+	image := resp.GetImage()
 
 	// Set the OMI ID in the state
-	ui.Message(fmt.Sprintf("OMI: %s", image.ImageId))
+	ui.Message(fmt.Sprintf("OMI: %s", image.GetImageId()))
 	omis := make(map[string]string)
-	omis[s.RawRegion] = image.ImageId
+	omis[s.RawRegion] = image.GetImageId()
 	state.Put("omis", omis)
 
 	// Wait for the image to become ready
 	ui.Say("Waiting for OMI to become ready...")
-	if err := osccommon.WaitUntilOscImageAvailable(oscconn, image.ImageId); err != nil {
+	if err := osccommon.WaitUntilOscImageAvailable(oscconn, *image.ImageId); err != nil {
 		log.Printf("Error waiting for OMI: %s", err)
-		imagesResp, _, err := oscconn.ImageApi.ReadImages(context.Background(), &osc.ReadImagesOpts{
-			ReadImagesRequest: optional.NewInterface(osc.ReadImagesRequest{
-				Filters: osc.FiltersImage{
-					ImageIds: []string{image.ImageId},
-				},
-			}),
-		})
+		req := oscgo.ReadImagesRequest{
+			Filters: &oscgo.FiltersImage{ImageIds: &[]string{image.GetImageId()}},
+		}
+		imagesResp, _, err := oscconn.Api.ImageApi.ReadImages(oscconn.Auth).ReadImagesRequest(req).Execute()
 		if err != nil {
 			log.Printf("Unable to determine reason waiting for OMI failed: %s", err)
 			err = fmt.Errorf("Unknown error waiting for OMI")
 		} else {
-			stateReason := imagesResp.Images[0].StateComment
-			err = fmt.Errorf("Error waiting for OMI. Reason: %s", stateReason)
+			stateReason := imagesResp.GetImages()[0].GetStateComment()
+			err = fmt.Errorf("Error waiting for OMI. Reason: %s", *stateReason.StateMessage)
 		}
 
 		state.Put("error", err)
 		ui.Error(err.Error())
 		return multistep.ActionHalt
 	}
 
-	imagesResp, _, err := oscconn.ImageApi.ReadImages(context.Background(), &osc.ReadImagesOpts{
-		ReadImagesRequest: optional.NewInterface(osc.ReadImagesRequest{
-			Filters: osc.FiltersImage{
-				ImageIds: []string{image.ImageId},
-			},
-		}),
-	})
+	req := oscgo.ReadImagesRequest{
+		Filters: &oscgo.FiltersImage{ImageIds: &[]string{image.GetImageId()}},
+	}
+	imagesResp, _, err := oscconn.Api.ImageApi.ReadImages(oscconn.Auth).ReadImagesRequest(req).Execute()
 	if err != nil {
 		err := fmt.Errorf("Error searching for OMI: %s", err)
 		state.Put("error", err)
 		ui.Error(err.Error())
 		return multistep.ActionHalt
 	}
-	s.image = &imagesResp.Images[0]
+	s.image = &imagesResp.GetImages()[0]
 
 	snapshots := make(map[string][]string)
-	for _, blockDeviceMapping := range imagesResp.Images[0].BlockDeviceMappings {
-		if blockDeviceMapping.Bsu.SnapshotId != "" {
-			snapshots[s.RawRegion] = append(snapshots[s.RawRegion], blockDeviceMapping.Bsu.SnapshotId)
+	blockMapping := imagesResp.GetImages()[0].BlockDeviceMappings
```

Comment:
`GetBlockDeviceMappings()`, this is usefull because if the var is  null, the getter will return an empty array see [link](https://github.com/outscale/osc-sdk-go/blob/7a11a0906c025139fb2b6f25cbd8f38890121333/v2/model_read_images_response.go#L43))